doc/developer/platform/ux.md

```diff
@@ -1,20 +1,945 @@
+evaluated in the above proposal.
+
+### Why are regions siloed?
+
+The ideal user experience would do away with the region silo. Let clusters
+access sources from any region. Let clusters span regions and failover between
+regions.
+
+But sharing infrastructure across regions causes high latency, high network
+transfer fees, and introduces hard distributed systems problems. We don't have a
+good handle on how to solve those problems. Colocating the important
+infrastructure in a single region eliminates all these problems.
+
+Put another way: we are not confident we can deliver a cross-region Materialize
+Platform, but we are confident we can deliver a single-region Materialize
+Platform.
+
+### Why clusters?
+
+The ideal user experience would not involve clusters. Users would show up with
+a set of queries, and Materialize Platform would automatically schedule those
+queries in clusters.
+
+This is absolutely the long-term vision for Materialize Platform, but we don't
+yet know how to make it happen.
+
+The problem is somewhat analogous to automatic building of indexes in
+traditional databases. The ideal user experience would not involve picking which
+indexes to build; it would all be handled by the database system automatically.
+Much research has gone into this problem, and yet in practice developers and
+DBAs still manually create indexes based on the observed performance of the
+system.
+
+### Why are sources, views, et al. shared between clusters?
+
+The general principle is that everything that can be shared *is* shared, to
+allow users maximum flexibility where there are no technical blockers. Users can
+then apply access control restrictions via [RBAC](#role) to opt-out of sharing
+when required.
+
+### Upgrade cadence
+
+A common request from customers is the ability to control when upgrades happen
+or to opt out of upgrades entirely. Honoring these requests is not in our best
+interests. Weekly upgrades are critical for our engineering velocity:
+
+  * Having only two supported versions massively reduces the state space of the
+    software running in production, which reduces bugs. The CLOUD OPERATOR
+    doesn't need to handle subtly different behavior in the dozens of deployed
+    Materialize Platform; it can limit itself to the most recent two versions.
+
+  * Migrating through breaking changes requires only two weeks.
+
+    Suppose we want to migrate our on-disk metadata format. In week one, we
+    release a version that is capable of reading the old format but rewrites all
+    data in the new format. In week two, we release a version that can only read
+    the new format, and delete the code that reads the old format. The tech debt
+    (reading the old format) only lives for two weeks.
+
+    If we were to instead allow customers to stay on an old version for 6-12
+    months, we'd be stuck maintaining the code that reads the old format for
+    another 12-24 months!
```

Comment:
Having worked on desktop software before, I'm intimately familiar with how this can be bad so I'm very happy to hear the plan is to limit old versions!